src/tests/system/tests/test_infopipe.py

```diff
@@ -0,0 +1,234 @@
+    assert uid == 10001, "user.Get(user_path, 'uidNumber') returned an unexpected UID"
+
+    gid = user.gidNumber
+    assert gid == 20001, "user.gidNumber returned an unexpected GID"
+
+    props = user.GetAll("org.freedesktop.sssd.infopipe.Users.User")
+    assert props["uidNumber"] == 10001, "GetAll(Users.User) returned an unexpected uidNumber"
+    assert props["gidNumber"] == 20001, "GetAll(Users.User) returned an unexpected gidNumber"
+    assert props["name"] == "user-1", "GetAll(Users.User) returned an unexpected name"
+    assert props["homeDirectory"] == "/home/user-1", "GetAll(Users.User) returned an unexpected homeDirectory"
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_infopipe__get_domain_properties(client: Client, provider: GenericProvider):
+    """
+    :title: Access a domain's information through InfoPipe
+    :setup:
+        1. Launch SSSD
+    :steps:
+        1. Call IsOnline()
+        2. Call GetAll() with "org.freedesktop.sssd.infopipe.Domains"
+        3. Read the property enumerable
+    :expectedresults:
+        1. Domain is online
+        2. All data is returned
+        3. Is enumerable
+    :customerscenario: False
+    """
+    client.sssd.start()
+
+    domain = client.ifp.getObject("/org/freedesktop/sssd/infopipe/Domains/test")
+
+    res = domain.IsOnline()
+    assert isinstance(res, bool), "IsOnline() didn't return a boolean"
+    assert res, "IsOnline() returned false"
+
+    res = domain.GetAll("org.freedesktop.sssd.infopipe.Domains")
+    assert isinstance(res, dict), "GetAll(infopipe.Domains) didn't return a dict"
+    assert res["name"] == "test", "GetAll(infopipe.Domains) returned an unexpected name"
+    assert res["provider"] == "ldap", "GetAll(infopipe.Domains) returned an unexpected provider"
+    assert (
+        res["login_format"] == r"^((?P<name>.+)@(?P<domain>[^@]+)|(?P<name>[^@]+))$"
+    ), "GetAll(infopipe.Domains) returned an unexpected login_format"
+    assert isinstance(
+        res["primary_servers"], list
+    ), "GetAll(infopipe.Domains) returned a primary_servers which is not a list"
+    assert (
+        len(res["primary_servers"]) == 1
+    ), "GetAll(infopipe.Domains) returned a wrong number of elements in primary_servers"
+    assert (
+        res["primary_servers"][0] == "ldap://master.ldap.test"
+    ), "GetAll(infopipe.Domains) returned a wrong URI in primary_servers"
+
+    res = domain.enumerable
+    assert isinstance(res, bool), "Property enumerable didn't return a boolean"
+    assert not res, "Property enumerable returned the wrong value"
+
+
+@pytest.mark.ticket(gh=6020, bz=2128840, jira="SSSD-5054")
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_infopipe__list_by_attr(client: Client, provider: GenericProvider):
+    """
+    :title: Test InfoPipe.Users.ListByAttr()
+    :setup:
+        1. Add 'user-1' to the provider
+        2. Add 'user-2' to the provider
+        3. Add 'user-3' to the provider
```

Comment:
Hi,

the old test also covered users from an "application" domain, i.e. a non-POSIX domain. Did you skip those because the framework currently does not support creating non-POSIX users or is there a different reason?

bye,
Sumit

Reply:
I wrote this test so long ago I now no longer remember the reason for this change.

Any way, I re-added the non-POSIX domain part back and it's working.

Reply:
Hi,

thank you, can you mention the app domain in the `:setup` and the `:expectedresults` of the test description as well.

bye,
Sumit

Reply:
Done.

Reply:
Thanks.